baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/IllegalSafeLoggingArgument.java

```diff
@@ -497,4 +499,67 @@ private Safety getLambdaRequiredReturnSafety(LambdaExpressionTree tree, VisitorS
         }
         return SafetyAnnotations.getSafety(returnType.type(), state);
     }
+
+    @Override
+    public Description matchMemberReference(MemberReferenceTree tree, VisitorState state) {
+        // This is the type the reference gets "cast" into, whether through assignment, return type, argument, etc
+        Type castType = state.getTypes().findDescriptorType(ASTHelpers.getType(tree));
```

Comment:
Should we check if `ASTHelpers.getType(tree)` returns null or `findDescriptorType` returns null and exit early?

Reply:
Good call!